src/gui/mzroll/downloadmanager.cpp

```diff
@@ -0,0 +1,86 @@
+#include "downloadmanager.h"
+#include <QNetworkAccessManager>
+#include <QNetworkRequest>
+#include  <QNetworkReply>
+#include <QDebug>
+#include "pollyintegration.h"
+
+DownloadManager::DownloadManager():
+    _manager(new QNetworkAccessManager),
+    _reply(nullptr),
+    err(false)
+{
+}
+
+void DownloadManager::setRequest(const QString &url, void *requester, bool async)
```

Comment:
Is the `requester` not being lost in this method? Maybe we can remove it if not needed.

Reply:
`requester` can be anyone. For now it's just `PollyIntegration`, so it may seem like there's no point of `requester` but in future we might have more requesters. 
How will more `requesters` be handled that is still something that needs to be figured out.

Reply:
So multiple objects could request through a single DownloadManager? I was under the impression that one such object would be used for only one request. In case of multiple requesting clients, we would have to have multiple data values too, since those requests could be made at different points of time so the data returned each time (for the same URL) could be different.

For now, we can skip all these considerations, given our time constraint. Resolving this conversation.